manifests/job.pp

```diff
@@ -34,11 +37,16 @@
   $jobname  = $title,
   $enabled  = 1,
   $ensure   = 'present',
+  $folder   = undef,
   $difftool = '/usr/bin/diff -b -q',
 ){
   include ::jenkins::cli
 
-  validate_string($difftool)
+  validate_string($difftool, $folder)
+
+  if $folder and (!defined(Jenkins::Plugin['cloudbees-folder'])) {
```

Comment:
It would be better to use the `jenkins_plugins` fact here:

    if $folder and (empty(grep([ $::jenkins_plugins ], 'cloudbees-folder'))) {